railties/lib/rails/generators/devcontainer.rb

```diff
@@ -84,128 +78,180 @@ def devcontainer_needs_redis?
           !(options.skip_action_cable? && options.skip_active_job?)
         end
 
-        def db_port_for_devcontainer(database = options[:database])
-          case database
-          when "mysql", "trilogy" then 3306
-          when "postgresql"       then 5432
-          end
+        def devcontainer_database
+          @devcontainer_database ||= Database.build(options[:database])
         end
 
-        def db_name_for_devcontainer(database = options[:database])
-          case database
-          when "mysql"          then "mysql"
-          when "trilogy"        then "mariadb"
-          when "postgresql"     then "postgres"
-          end
+        def devcontainer_db_service_yaml(**options)
+          return unless service = devcontainer_database.service
+
+          { devcontainer_database.name => service }.to_yaml(**options)[4..-1]
```

Comment:
What does 4 and 1 means here?

Reply:
It's the leading `---` that separate the body of the doc from directives. I don't know to_yaml takes an option to remove it.